modules/web/src/com/haulmont/cuba/web/gui/components/WebSplitPanel.java

```diff
@@ -356,34 +358,44 @@ protected Unit convertLegacyUnit(int unit) {
     @Override
     public void setDockable(boolean dockable) {
         if (orientation == SplitPanel.ORIENTATION_VERTICAL) {
-            throw new IllegalStateException("Docking is not available for the vertically oriented SplitPanel.");
+            ((CubaVerticalSplitPanel) component).setDockable(dockable);
+        } else {
+            ((CubaHorizontalSplitPanel) component).setDockable(dockable);
         }
-        ((CubaHorizontalSplitPanel) component).setDockable(dockable);
     }
 
     @Override
     public boolean isDockable() {
         if (orientation == SplitPanel.ORIENTATION_VERTICAL) {
-            return false;
+            return ((CubaVerticalSplitPanel) component).isDockable();
         }
         return ((CubaHorizontalSplitPanel) component).isDockable();
     }
 
     @Override
     public void setDockMode(DockMode dockMode) {
+        SplitPanelDockMode mode = SplitPanelDockMode.valueOf(dockMode.name());
         if (orientation == SplitPanel.ORIENTATION_VERTICAL) {
-            throw new IllegalStateException("Docking is not available for the vertically oriented SplitPanel.");
+            if (mode == SplitPanelDockMode.LEFT || mode == SplitPanelDockMode.RIGHT) {
```

Comment:
Move checks to corresponding Vaadin components (`CubaVerticalSplitPanel` and `CubaHorizontalSplitPanel`), because these components accept `SplitPanelDockMode` with all available directions that can be unsupported be a certain component.